pkg/resource/comparator.go

```diff
@@ -95,3 +102,102 @@ func (c *comparatorBuilder) BuildAsFunc() (reflect.Type, func(deployed resource.
 	c.Build()
 	return c.comparator.ResourceType, c.comparator.CompFunc
 }
+
+func containAllLabels(deployed resource.KubernetesResource, requested resource.KubernetesResource) bool {
+	deployedLabels := deployed.GetLabels()
+	requestedLabels := requested.GetLabels()
+
+	for key, value := range requestedLabels {
+		if deployedLabels[key] != value {
+			return false
+		}
+	}
+
+	return true
+}
+
+// CreateDeploymentConfigComparator creates a new comparator for DeploymentConfig using Trigger and RollingParams
+func CreateDeploymentConfigComparator() func(deployed resource.KubernetesResource, requested resource.KubernetesResource) bool {
+	return func(deployed resource.KubernetesResource, requested resource.KubernetesResource) bool {
+		dc1 := deployed.(*appsv1.DeploymentConfig)
+		dc2 := requested.(*appsv1.DeploymentConfig).DeepCopy()
+
+		for i := range dc1.Spec.Triggers {
+			if len(dc2.Spec.Triggers) <= i {
+				return false
+			}
+			trigger1 := dc1.Spec.Triggers[i]
+			trigger2 := dc2.Spec.Triggers[i]
+			if trigger1.ImageChangeParams != nil && trigger2.ImageChangeParams != nil && trigger2.ImageChangeParams.From.Namespace == "" {
+				//This value is generated based on image stream being found in current or openshift project:
+				trigger1.ImageChangeParams.From.Namespace = ""
+			}
+		}
+
+		if dc2.Spec.Strategy.RollingParams == nil && dc1.Spec.Strategy.Type == dc2.Spec.Strategy.Type {
+			dc1.Spec.Strategy.RollingParams = dc2.Spec.Strategy.RollingParams
+		}
+		return true
+	}
+}
+
+// CreateBuildConfigComparator creates a new comparator for BuildConfig using Label, Trigger and SourceStrategy
+func CreateBuildConfigComparator() func(deployed resource.KubernetesResource, requested resource.KubernetesResource) bool {
+	return func(deployed resource.KubernetesResource, requested resource.KubernetesResource) bool {
+		bc1 := deployed.(*buildv1.BuildConfig)
+		bc2 := requested.(*buildv1.BuildConfig).DeepCopy()
+
+		if !containAllLabels(bc1, bc2) {
+			return false
+		}
+		if bc1.Spec.Strategy.SourceStrategy != nil {
```

Comment:
Won't worth sending a PR to utils covering this logic? (and the triggers as well).

Reply:
Same reason for BC: https://github.com/kiegroup/kie-cloud-operator/blob/acce7c16d06ea419fa9fdb37c385f721c4b7fa7c/pkg/controller/kieapp/kieapp_controller.go#L233